plugins/utils/navbuddy.nix

```diff
@@ -0,0 +1,175 @@
+            border = helpers.defaultNullOpts.mkBorder "rounded" "double" "solid" "none" null ''
+              "rounded", "double", "solid", "none"  or an array with eight chars building up the border in a clockwise fashion
+              starting with the top-left corner. eg: { "╔", "═" ,"╗", "║", "╝", "═", "╚", "║" }.
+            '';
+          };
+
+          mid = {
+            size = helpers.defaultNullOpts.mkStr "40%";
+            border = helpers.defaultNullOpts.mkBorder "rounded" "double" "solid" "none" null ''
+              "rounded", "double", "solid", "none"  or an array with eight chars building up the border in a clockwise fashion
+              starting with the top-left corner. eg: { "╔", "═" ,"╗", "║", "╝", "═", "╚", "║" }.
+            '';
+          };
+
+          right = {
+            border = helpers.defaultNullOpts.mkBorder "rounded" "double" "solid" "none" null ''
+              "rounded", "double", "solid", "none"  or an array with eight chars building up the border in a clockwise fashion
+              starting with the top-left corner. eg: { "╔", "═" ,"╗", "║", "╝", "═", "╚", "║" }.
+            '';
+            preview = helpers.defaultNullOpts.mkEnum ["leaf" "always" "never"] "leaf" ''
+              Right section can show previews too.
+                    Options: "leaf", "always" or "never"
+            '';
+          };
+        };
+      };
+
+      nodeMarkers = {
+        enabled = true;
+        icons = {
+          leaf = helpers.defaultNullOpts.mkStr "  " ''
+            The icon to use for leaf nodes.
+          '';
+
+          leaf_selected = helpers.defaultNullOpts.mkStr " → " ''
+            The icon to use for selected leaf node.
+          '';
+          branch = helpers.defaultNullOpts.mkStr "  " ''
+            The icon to use for branch nodes.
+          '';
+        };
+      };
+
+      icons =
+        mapAttrs
+        (
+          name: default:
+            default "icon for ${name}."
+        )
+        {
+          File = "󰈙 ";
+          Module = " ";
+          Namespace = "󰌗 ";
+          Package = " ";
+          Class = "󰌗 ";
+          Method = "󰆧 ";
+          Property = " ";
+          Field = " ";
+          Constructor = " ";
+          Enum = "󰕘";
+          Interface = "󰕘";
+          Function = "󰊕 ";
+          Variable = "󰆧 ";
+          Constant = "󰏿 ";
+          String = "󰀬 ";
+          Number = "󰎠 ";
+          Boolean = "◩ ";
+          Array = "󰅪 ";
+          Object = "󰅩 ";
+          Key = "󰌋 ";
+          Null = "󰟢 ";
+          EnumMember = " ";
+          Struct = "󰌗 ";
+          Event = " ";
+          Operator = "󰆕 ";
+          TypeParameter = "󰊄 ";
+        };
+
+      useDefaultMapping = helpers.defaultNullOpts.mkBool true ''
+        If set to false, only mappings set by user are set. Else default mappings are used for keys that are not set by user
+      '';
+
+      # TODO: how to set mappings, like in nvim-cmp ?
+      mapping = {};
```

Comment:
How do I set the typing for key maps ? 
For example we have: 
```nix
 mappings = {
        ["<esc>"] = actions.close(),        -- Close and cursor to original location
        ["q"] = actions.close(),

        ["j"] = actions.next_sibling(),     -- down
}
```

Reply:
You would map the keys to some actions (i.e. strings: `close`; `next_sibling`) and then you create the mappings from that yourself.
See https://github.com/nix-community/nixvim/blob/32a64af2318beac3c89d4d1fa0b73ce237c5cf1e/plugins/utils/spider.nix#L54-L65 as an example.